decayers/ERElasticScattering.cxx

```diff
@@ -103,16 +107,30 @@ Bool_t ERElasticScattering::Init() {
     return kFALSE;
   }
 
-  SetProjectileIonMass(fInputIonPDG->Mass());
-  SetTargetIonMass(fTargetIonPDG->Mass());
+  DefineOfIonsMasses();
+  LOG(DEBUG) << "ERElasticScattering::Init()" << FairLogger::endl;
+  LOG(DEBUG) << "Traget Ions Mass is " << GetTargetIonMass()
+             << ", Prigectile Ions Mass is " << GetProjectileIonMass() << FairLogger::endl;
 
-  if (fThetaLabRangeIsSet)
+  if (!fRelativisticMode) {
     ThetaRangesLab2CM(GetProjectileIonMass(), GetTargetIonMass());
-
-  return ThetaCDFRead();
+    if (!ThetaCDFRead()) {
+      LOG(FATAL) << "The input file which contains the CDF function can't be read!" << FairLogger::endl;
+      return kFALSE;
+    }
+  }
+  return kTRUE;
 }
 
 Bool_t ERElasticScattering::Stepping() {
+  if (fRelativisticMode && gMC->TrackPid() == fInputIonPDG->PdgCode()) {
+    ThetaRangesLab2CMRelativistic();
+    if (!ThetaCDFRead()) {
+      LOG(FATAL) << "The input file which contains the CDF function can't be read!" << FairLogger::endl;
```

Comment:
Move it to Init method

Reply:
I can't move It.
In this method ThetaRangesLab2CMRelativistic(); projectile momentum is used. 
To get this momentum I have to use 
 **TLorentzVector curLV;
  gMC->TrackMomentum(curLV);**

There is a problem that in Init() method gMC isn't accessible for use. 
The first point of access to gMC is in the Stepping() method.

I think, the solution that I chose for this problem isn't good. But unfortunately, I had only this ideas to solve It.
Can you give me some advises? How should I solve this problem?